### PR TITLE
[SF2035] Change processing user data for the current project

### PR DESCRIPTION
**Pull request by arsentur**
**Description:**
- Change processing user data for the current project

**YouTrack:**
[SF2035-5](https://youtrack.raccoongang.com/issue/SF2035-5)

### CODE REVIEW

File: edx_oauth_client/pipeline.py [hunk 1]
```diff
@@ -38,13 +38,18 @@ def ensure_user_information(
 
         # Received fields could be pretty different from the expected, mandatory are only 'username' and 'email'
         data['username'] = user_data.get('username', user_data.get('name'))
-        data['first_name'] = user_data.get('firstName', user_data.get('first_name'))
-        data['last_name'] = user_data.get('lastName', user_data.get('last_name'))
+        data['first_name'] = user_data.get('firstname', user_data.get('first_name'))
+        data['second_name'] = user_data.get('secondname', '')
```
Comment (NikolayBorovenskiy) [inline edx_oauth_client/pipeline.py]: Did not get the case here `user_data.get('secondname', '')` and the line above you have this `user_data.get('firstname', user_data.get('first_name'))`
As I see it should be consistent thus `user_data.get('secondname', user_data.get('second_name'))`

File: edx_oauth_client/pipeline.py [hunk 2]
```diff
@@ -38,13 +38,18 @@ def ensure_user_information(
...
         data['email'] = user_data.get('email')
         data['country'] = country
         data['access_token'] = access_token
+
+        if user_data.get('gender'):
+            data['gender'] = user_data.get('gender')[0]
+
         if any((data['first_name'], data['last_name'])):
-            data['name'] = u'{} {}'.format(data['first_name'], data['last_name']).strip()
+            data['name'] = '{} {} {}'.format(['first_name'], data['second_name'], data['last_name']).strip()
```
Comment (NikolayBorovenskiy) [inline edx_oauth_client/pipeline.py]: Is it because Ironwood has `python 3` on a board?

File: edx_oauth_client/pipeline.py [hunk 3]
```diff
@@ -38,13 +38,18 @@ def ensure_user_information(
...
-        data['last_name'] = user_data.get('lastName', user_data.get('last_name'))
+        data['first_name'] = user_data.get('firstname', user_data.get('first_name'))
+        data['second_name'] = user_data.get('secondname', '')
+        data['last_name'] = user_data.get('lastname', user_data.get('last_name'))
         data['email'] = user_data.get('email')
         data['country'] = country
         data['access_token'] = access_token
+
+        if user_data.get('gender'):
+            data['gender'] = user_data.get('gender')[0]
```
Comment (NikolayBorovenskiy) [inline edx_oauth_client/pipeline.py:56]: Please explain the meaning of index `[0]` usage?

**Review by NikolayBorovenskiy — CHANGES_REQUESTED** [review]
In general good but I need a few explanations 🙂

File: edx_oauth_client/pipeline.py (hunk 2 above)
Comment (NikolayBorovenskiy) [reply edx_oauth_client/pipeline.py]: To the case `'{} {} {}'` I prefer to do at least this `'{0} {1} {2}'` It is more explicite as python's dzen.

File: edx_oauth_client/pipeline.py (hunk 3 above)
Comment (arsentur) [reply edx_oauth_client/pipeline.py:56]: Wrote comment above

File: edx_oauth_client/pipeline.py [hunk 4]
```diff
@@ -39,17 +40,23 @@ def ensure_user_information(
...
+            data['year_of_birth'] = datetime.datetime.strptime(date_of_birth, '%d.%m.%Y').year
+
+        # User API provide two possible variants "male" or "female", with are strings.
+        # We need extract only first letter and save it to the user profile model.
         if user_data.get('gender'):
             data['gender'] = user_data.get('gender')[0]
 
         if any((data['first_name'], data['last_name'])):
-            data['name'] = '{} {} {}'.format(['first_name'], data['second_name'], data['last_name']).strip()
+            data['name'] = u'{} {}'.format(data['first_name'], data['last_name']).strip()
```
Comment (NikolayBorovenskiy) [inline edx_oauth_client/pipeline.py:59]: Why did you bring back `u''`? If Ironwood is built on python 3, that it does not need already.

**Review by NikolayBorovenskiy — COMMENTED** [review]
Just one comment.

File: edx_oauth_client/pipeline.py (hunk 4 above)
Comment (arsentur) [reply edx_oauth_client/pipeline.py:59]: Now ironwood release uses python 2.7

**Comment by NikolayBorovenskiy** [comment]
Great. No objections from my side. You get my approval.)

**Review by NikolayBorovenskiy — APPROVED** [review]

**Review by idegtiarov — APPROVED** [review]

**Pull request merged**